### PR TITLE
Update go-git to v5

### PR DESCRIPTION
**Pull request by klaidliadon**
This PR updates the `src-d/go-git` dependency, which is archived in favor of `go-git/go-git`.
The current version suffers from [this vulnerability](https://github.com/go-git/go-git/security/advisories/GHSA-449p-3h89-pw88).

### CODE REVIEW

File: internal/localfs/localfs.go [hunk 1]
```diff
@@ -8,11 +8,11 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/go-git/go-billy/v5/osfs"
+	git "github.com/go-git/go-git/v5"
```
Comment (VojtechVitek) [inline internal/localfs/localfs.go:12]: Screen Shot 2024-05-20 at 10 52 06 AM

**Review by VojtechVitek — APPROVED** [review]
LGTM

**Comment by posener** [comment]
Hey, I pushed this update to master. Sorry for not merging this commit, I wanted to run the `go mod tidy` command myself (there are a lot of cryptic updates in the `go.sum` file.

**Comment by VojtechVitek** [comment]
@posener great, thank you!

@klaidliadon would you mind fixing the CI (by upgrading Go versions at https://github.com/posener/gitfs/blob/master/.github/workflows/test.yml#L11-L13), so Eyal can release a new `git tag`, please?

**Pull request closed without merging**